src/routes/auth.js

```diff
@@ -0,0 +1,86 @@
+//required packages
+const express = require("express");
+const jwt = require('jsonwebtoken');
+const bcrypt = require("bcrypt");
+const fileWriter  = require("../helper/file-helper");
+
+const validator = require("../helper/validator");
+//Tasks Data
+const usersData = require("../data/users.json");
+
+//router initialization
+const router = express.Router();
+
+//API to register
+
+router.post("/register", (req, res) => {
+  try {
+    let userDetails = req.body;
+    let { users } = usersData;
+    let validatorObj = validator.validateUserInfo(userDetails, users);
+    console.log("validatorObj",validatorObj, users, userDetails);
+    if (validatorObj.status) {
+      userDetails.password = bcrypt.hashSync(userDetails.password, 8);
+      let result = fileWriter(userDetails, users);
+      if (result.status) {
+        res.status(200).send(result);
+      } else {
+        res.status(500).send(result);
+      }
+    } else {
+      res.status(400);
+      res.send(validatorObj);
+    }
+  } catch (error) {
+    res.status(500).send({ message: "Something went wrong !" });
+  }
+});
+
+//API to Login
+router.post("/login", (req, res) => {
+  try {
+    let currUser = req.body;
```

Comment:
Also destructure the objects where ever required. For example -:

const {email, password} = currentUser;